openedx/core/djangoapps/programs/utils.py

```diff
@@ -151,7 +193,12 @@ def progress(self, programs=None, count_only=True):
                 if self._is_course_complete(course):
                     completed.append(course)
                 elif self._is_course_in_progress(course):
-                    in_progress.append(course)
+                    course_in_progress = self.does_course_still_qualify_as_in_progress(now, enrolled_run_modes, course)
+                    if course_in_progress:
+                        in_progress.append(course)
+                    else:
+                        course['expired'] = True
```

Comment:
This will modify any programs provided as arguments in place. Functions/methods should either mutate their arguments or return something, but not both. Is there a less surprising way to do this?